File: Parenthesis_checker/code.cpp
class Solution
{
    public:
    //Function to check if brackets are balanced or not.
    bool ispar(string x)
    {
        // Your code here
        string res;
        char temp;
        for(int i=0; i<x.size(); i++){
            if(res.size() == 0){
                temp = x[i];
                res.append(1, temp);
            }
            else if(x[i] == '(' || x[i] == '[' || x[i] == '{'){
                temp = x[i];
                res.append(1, temp);
            } 
            else if(x[i] == ')'){
                if(res[res.size()-1] == '('){
                    res.pop_back();
                }
                else{
                    temp = x[i];
                    res.append(1, temp);
                }
            }
            else if(x[i] == ']'){
                if(res[res.size()-1] == '['){
                    res.pop_back();
                }
                else{
                    temp = x[i];
                    res.append(1, temp);
                }
            }
            else if(x[i] == '}'){
                if(res[res.size()-1] == '{'){
                    res.pop_back();
                }
                else{
                    temp = x[i];
                    res.append(1, temp);
                }
            }
        }
        if(res.size() == 0){
            return 1;
        }
        else{
            return 0;
        }
    }

};

File: Parenthesis_checker/code.java
class Solution
{
    //Function to check if brackets are balanced or not.
    static boolean ispar(String x)
    {
        Stack<Character> s=new Stack<>();
        s.push(x.charAt(0));
        for(char i:x.substring(1).toCharArray()){
            if(i=='(' || i=='{' || i=='['){
                s.push(i);
                continue;
            }
            if(s.size()!=0){
                if((s.peek()=='(' && i==')') || (s.peek()=='{' && i=='}') || (s.peek()=='[' && i==']')){
                    s.pop();
                }
                else{
                    return false;
                }
            }
            else{
                return false;
            }
            
        }
        if(s.size()!=0){
            return false;
        }
        return true;
    }
}

